Src/EB/AMReX_EB_levelset.cpp

```diff
@@ -157,18 +192,21 @@ void LSFactory::fill_valid_kernel(){
 #pragma omp parallel
 #endif
     for(MFIter mfi( * ls_grid, true); mfi.isValid(); ++ mfi) {
-        Box tile_box = mfi.tilebox();
-        const int * lo = tile_box.loVect();
-        const int * hi = tile_box.hiVect();
+        Box tile_box                          = mfi.tilebox();
+        Array4<Real const> const & ls_tile    = ls_grid->array(mfi);
+        Array4<int>        const & valid_tile = ls_valid->array(mfi);
 
-        const auto & ls_tile = (* ls_grid)[mfi];
-        auto & valid_tile    = (* ls_valid)[mfi];
+        ParallelFor(tile_box,
+                [=] AMREX_GPU_DEVICE (int i, int j, int k) {
 
-        amrex_eb_fill_valid(lo, hi,
-                            BL_TO_FORTRAN_3D(valid_tile),
-                            BL_TO_FORTRAN_3D(ls_tile),
-                            & search_radius);
+                    bool valid_cell = neighbour_is_valid(
+                            tile_box, ls_tile, i, j, k, search_radius
+                        );
 
+                    if (neighbour_is_valid) valid_tile(i, j, k) = 1;
```

Comment:
`if (valid_cell)` ?

Reply:
Ooops -- yes `if (valid_cell)`. I wonder why that even worked